code/modules/xenoarchaeology/xenoartifact.dm

```diff
@@ -0,0 +1,470 @@
+/obj/item/xenoartifact
+	name = "artifact"
+	icon = 'icons/obj/xenoarchaeology/xenoartifact.dmi'
+	icon_state = "map_editor"
+	w_class = WEIGHT_CLASS_NORMAL
+	light_color = LIGHT_COLOR_FIRE
+	desc = "A strange alien device. What could it possibly do?"
+	throw_range = 3
+
+	///How much input the artifact is getting from activator traits
+	var/charge = 0
+	///This isn't a requirement anymore. This just affects how effective the charge is 
+	var/charge_req 
+	///Processing type, used for tick
+	var/process_type
+	///List of targted entities for traits
+	var/list/true_target = list() 
+
+	///Associated traits & colour
+	var/material 
+	///activation trait, minor 1, minor 2, minor 3, major, malfunction
+	var/list/traits = list()
+	///Touch hint
+	var/datum/xenoartifact_trait/touch_desc
+	///used for special examine circumstance, science goggles & ghosts
+	var/special_desc = "The artifact is made from a"
+	///Description used for label, used because directly adding shit to desc isn't a good idea
+	var/label_desc
+	///How far the artifact can reach
+	var/max_range = 1
+
+	//Used for signaler trait
+	var/code 
+	var/frequency
+	var/datum/radio_frequency/radio_connection
+
+	//Time between uses
+	var/cooldown = 8 SECONDS
+	///Extra time traits can add to the cooldown
+	var/cooldownmod = 0
+	COOLDOWN_DECLARE(xenoa_cooldown)
+
+	//Associated with random sprite stuff. It's setup as [4] so it's easier to check for slots being assigned by traits and such
+	var/list/icon_slots[4]
+	var/mutable_appearance/icon_overlay 
+
+	///Everytime the artifact is used this increases. When this is successfully proc'd the artifact gains a malfunction and this is lowered.
+	var/malfunction_chance = 0
+	///How much the chance can change in a sinlge itteration
+	var/malfunction_mod = 0.1
+
+	//snowflake variable for shaped
+	var/transfer_prints = FALSE
+
+/obj/item/xenoartifact/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/xenoartifact_pricing)
+	AddComponent(/datum/component/discoverable, XENOA_DP, TRUE) //Same values as original artifacts from exploration
+
+/obj/item/xenoartifact/Initialize(mapload, difficulty)
+	. = ..()
+
+	material = difficulty //Difficulty is set, in most cases
+	if(!material)
+		material = pick(XENOA_BLUESPACE, XENOA_PLASMA, XENOA_URANIUM, XENOA_BANANIUM) //Maint artifacts and similar situations
+
+	var/datum/component/xenoartifact_pricing/xenop = GetComponent(/datum/component/xenoartifact_pricing)
```

Comment:
Rather than get the component and manually set values, you should send a signal that is handled by the component to update the price.